core/src/main/java/org/eclipse/hono/util/TenantDataVolume.java

```diff
@@ -0,0 +1,126 @@
+
+    @JsonProperty(RegistryManagementConstants.FIELD_MAX_BYTES)
+    private long maxBytes = -1;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_PERIOD)
+    private TenantDataVolumePeriod period;
+
+    /**
+     * Gets the offsetDateTime on which the data volume limit came into effect.
+     *
+     * @return The offsetDateTime as an ISO string or 
+     *         {@code null} if not set.
+     */
+    @JsonGetter(RegistryManagementConstants.FIELD_EFFECTIVE_SINCE)
+    public String getEffectiveSinceAsString() {
+        return Optional.ofNullable(effectiveSince)
+                .map(offsetDateTime -> offsetDateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME))
+                .orElse(null);
+    }
+
+    /**
+     * Gets the offsetDateTime on which the data volume limit came into effect.
+     *
+     * @return The offsetDateTime on which the data volume limit came into effective or 
+     *         {@code null} if not set.
+     */
+    @JsonIgnore
+    public OffsetDateTime getEffectiveSince() {
+        return effectiveSince;
+    }
+
+    /**
+     * Sets the offsetDateTime on which the data volume limit came into effect.
+     * 
+     * @param effectiveSince the date string on which the data volume limit came into effect
+     *                       and it comply to the {@link DateTimeFormatter#ISO_OFFSET_DATE_TIME}.
+     */
+    @JsonSetter(RegistryManagementConstants.FIELD_EFFECTIVE_SINCE)
+    public void setEffectiveSince(final String effectiveSince) {
+        this.effectiveSince = parse(effectiveSince);
+    }
+
+    /**
+     * Gets the maximum number of bytes to be allowed for the time period defined by the
+     * {@link RegistryManagementConstants#FIELD_PERIOD_MODE} and 
+     * {@link RegistryManagementConstants#FIELD_PERIOD_NO_OF_DAYS}.
+     *
+     * @return The maximum number of bytes or {@link RegistryManagementConstants#DEFAULT_MAX_BYTES} 
+     *         if not set.
+     */
+    public Long getMaxBytes() {
+        return maxBytes;
+    }
+
+    /**
+     * Sets the maximum number of bytes to be allowed for the time period defined by the
+     * {@link RegistryManagementConstants#FIELD_PERIOD_MODE} and 
+     * {@link RegistryManagementConstants#FIELD_PERIOD_NO_OF_DAYS}.
+     *
+     * @param maxBytes The maximum number of bytes to be allowed.
+     */
+    public void setMaxBytes(final Long maxBytes) {
+        this.maxBytes = maxBytes;
+    }
+
+    /**
+     * Gets the period for the data usage calculation.
+     *
+     * @return The period for the data usage calculation.
+     */
+    public TenantDataVolumePeriod getPeriod() {
+        return period;
+    }
+
+    /**
+     * Sets the period for the data usage calculation.
+     *
+     * @param period The period for the data usage calculation.
+     */
+    public void setPeriod(final TenantDataVolumePeriod period) {
+        this.period = period;
+    }
+
+    private OffsetDateTime parse(final String timestamp) {
```

Comment:
I would strongly suggest to **not** use offsetted DateTime. This will only cause problems unless you have a real reason, which in most cases only exists using this in UI.

What you probably want is `Instant`, which is a specific point in time.

Reply:
I will change it to use `Instant`.